### PR TITLE
reorganize schema

### PR DESCRIPTION
**Pull request by eidan66**

### CODE REVIEW

File: server/controller/item.controller.js [hunk 1]
```diff
@@ -1,19 +1,86 @@
...
+  } = req.body;
+
+  // Build items object
+  const itemFields = {};
+  if (name) itemFields.name = name;
+  if (description) itemFields.description = description;
+  if (category) itemFields.category = category;
+  if (brand) itemFields.brand = brand;
+  if (color) {
+    itemFields.color = color.split(',').map((color) => color.trim());
```
Comment (yonatansnir) [inline server/controller/item.controller.js:39]: Perfect Idea!

File: server/models/Cart.js [hunk 2]
```diff
@@ -1,26 +1,40 @@
...
+    {
+      item: {
+        type: mongoose.Schema.Types.ObjectId,
+        ref: 'item',
+      },
+    },
+  ],
+  price: [
+    {
+      totalItemsPrice: {
```
Comment (yonatansnir) [inline server/models/Cart.js]: You don't calculate the total price on the DB. It's better to do it on the client-side. It's short and simple.

File: server/models/Item.js [hunk 3]
```diff
@@ -1,18 +1,83 @@
...
+  description: {
+    type: String,
+    required: true,
+  },
+  category: {
+    //   Men / Women / Kids etc...
+    type: String,
+    required: true,
+  },
+  brand: {
```
Comment (yonatansnir) [inline server/models/Item.js:16]: All Items that we sell are NEW 😉

File: server/models/User.js [hunk 4]
```diff
@@ -1,52 +1,64 @@
-const mongoose = require('mongoose')
+const mongoose = require('mongoose');
 const UserSchema = new mongoose.Schema({
-    fullName: {
+  fullName: [
```
Comment (yonatansnir) [inline server/models/User.js]: Why is it an array?

**Review by yonatansnir — COMMENTED** [review]
You've done an excellent job!

File: server/models/Cart.js (hunk 2 above)
Comment (eidan66) [reply server/models/Cart.js]: The idea was save all the parameters in DB and calc it at the end (client side ).
I want save all the information before the customer pay to us.

File: server/models/User.js (hunk 4 above)
Comment (eidan66) [reply server/models/User.js]: Was mistake, will fix it in the next commit action.

**Pull request merged**